extensions/amp-lightbox/1.0/component.js

```diff
@@ -195,3 +190,43 @@ function LightboxWithRef(
 const Lightbox = forwardRef(LightboxWithRef);
 Lightbox.displayName = 'Lightbox';
 export {Lightbox};
+
+/**
+ *
+ * @param {!LightboxDef.CloseButtonProps} props
+ * @return {PreactDef.Renderable}
+ */
+function CloseButton({close, customCloseButton = <ScreenReaderCloseButton />}) {
```

Comment:
Why not pass a Component type here instead? Wrapping `click` is inverted (e.g. not here). Seems more natural to me.

```suggestion
function CloseButton({close, as = ScreenReaderCloseButton}) {
```

Reply:
Good question. I had considered this but assumed that a given close button would be given as JSX rather than a functional component, to support this case:
```
<Lightbox closeButton={<button>close</button> />
```

Are you instead recommending the following API?
```
function MyCloseButton(props) {
  return <button {props}>close</button>
}

<Lightbox closeButtonAs={MyCloseButton} />
```

Reply:
Right. Can also set inline as a render prop:

```js
<Lightbox closeButtonAs={(props) => <button {...props}>close</button>} />
```

If we need another click handler, looks more intentional imo:

```js
<Lightbox
  closeButtonAs={({onClick, ...rest}) => (
    <button
      onClick={(e) => {
        onClick(e);
        //...everything else
      }}
      {...rest}
    >
      close
    </button>
  )}
/>
```

Reply:
Updated in [9293281](https://github.com/ampproject/amphtml/pull/33105/commits/92932818075135aa22c793c47bec93aaced6f972) and it does feel more natural and intentional, thanks for the suggestion! As a temporary bridge for the AMP layer, I also added the following to make it work:
```js
/** @override */
updatePropsForRendering(props) {
  props['closeButtonAs'] = () => props['closeButton'];
}
```

This unfortunately also means that the AMP layer button has to define its own `on="tap..."` close action (as currently it ends up ignoring the given `onClick`) in the meantime, differing from the Preact layer which does not. I suspect in cases where we do take selectors now, it's worth trying to move to this same pattern of passing the generated `Slot`s as functions rather than as renderables, and while we're at it, making sure they propagate any given props. But I also think that we should solve this all at once in a separate PR. WDYT? Is the above okay for the scope of this PR?

Reply:
This makes sense, and I am on board with splitting to a separate PR to update the pattern and make sure props are propagated.

Also I think it's fine that the AMP layer button defines its own `on=tap...` for now as this matches with current AMP layer convention.  And is this the same API as `0.1`?

Reply:
Yes, that's correct.